sources/packages/web/src/components/common/students/StudentFileUploads.vue

```diff
@@ -0,0 +1,206 @@
+                </div>
+                <div v-else>
+                  <span class="mr-4">
+                    <v-icon icon="fa:far fa-file-alt" size="20"></v-icon
+                  ></span>
+                  <span>{{ slotProps.data.fileName }}</span>
+                </div>
+              </template></Column
+            >
+          </DataTable>
+        </toggle-content>
+      </content-group>
+      <formio-modal-dialog
+        :max-width="730"
+        ref="fileUploadModal"
+        title="Upload file"
+        :formData="initialData"
+        formName="uploadStudentDocumentsAEST"
+      >
+        <template #actions="{ cancel, submit }">
+          <v-row class="m-0 p-0">
+            <v-btn color="primary" variant="outlined" @click="cancel"
+              >Cancel</v-btn
+            >
+            <check-permission-role :role="Role.StudentUploadFile">
+              <template #="{ notAllowed }">
+                <v-btn
+                  class="float-right"
+                  @click="submit"
+                  color="primary"
+                  variant="elevated"
+                  :disabled="notAllowed"
+                  >Upload now</v-btn
+                >
+              </template>
+            </check-permission-role>
+          </v-row>
+        </template>
+      </formio-modal-dialog>
+    </body-header-container>
+  </tab-container>
+</template>
+
+<script lang="ts">
+import { onMounted, ref, defineComponent } from "vue";
+import {
+  DEFAULT_PAGE_LIMIT,
+  FormIOForm,
+  PAGINATION_LIST,
+  LayoutTemplates,
+  Role,
+} from "@/types";
+import { StudentService } from "@/services/StudentService";
+import {
+  useFormatters,
+  useFileUtils,
+  ModalDialog,
+  useFormioUtils,
+  useSnackBar,
+} from "@/composables";
+import FormioModalDialog from "@/components/generic/FormioModalDialog.vue";
+import { AESTFileUploadToStudentAPIInDTO } from "@/services/http/dto/Student.dto";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  components: {
+    FormioModalDialog,
+    CheckPermissionRole,
+  },
+  emits: ["loaded"],
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+    canUploadFiles: {
+      type: Boolean,
+      required: true,
+    },
+    canDownloadFiles: {
+      type: Boolean,
+      required: true,
+    },
+    studentFileUploads: {
+      type: Array,
+      required: true,
+    },
+  },
+  setup(props: any, context) {
+    const fileUploadModal = ref({} as ModalDialog<FormIOForm | boolean>);
+    const { dateOnlyLongString } = useFormatters();
+    const snackBar = useSnackBar();
+    const fileUtils = useFileUtils();
+    const formioUtils = useFormioUtils();
+    const initialData = ref({ studentId: props.studentId });
+
+    const loadStudentFileUploads = async () => {
+      context.emit("loaded", props.studentId);
+    };
+
+    onMounted(async () => {
```

Comment:
suggestion, you can also do `onMounted(loadStudentFileUploads);`

Reply:
I feel its better to explicitly specify that `loadStudentFileUploads` is an async call.

Reply:
I would be inclined to agree with @ann-aot because the code looks cleaner.

Reply:
If the other suggestion about making the API call unique will be implemented, the "loaded" event will be removed, right?

Reply:
Yes